src/CoreFn/Expr.purs

```diff
@@ -57,6 +59,22 @@ derive instance eqLiteral :: Eq a => Eq (Literal a)
 derive instance genericLiteral :: Generic a => Generic (Literal a)
 derive instance ordLiteral :: Ord a => Ord (Literal a)
 
+instance foldableLiteral :: Foldable Literal where
+  foldMap f (ArrayLiteral x) = foldMap f x
+  foldMap f (ObjectLiteral x) = foldMap (foldMap f) x
+  foldMap _ _ = mempty
+
+  foldl f = foldlDefault f
+  foldr f = foldrDefault f
+
+instance functorLiteral :: Functor Literal where
+  map _ (NumericLiteral x) = NumericLiteral x
+  map _ (StringLiteral x) = StringLiteral x
+  map _ (CharLiteral x) = CharLiteral x
+  map _ (BooleanLiteral x) = BooleanLiteral x
```

Comment:
You should now be able to use `derive instance functorLiteral :: Functor Literal` :)